src/core/thread/network_data_local.cpp

```diff
@@ -344,8 +344,6 @@ void Local::UpdateRloc(void)
             break;
         }
     }
-
-    ClearResubmitDelayTimer();
```

Comment:
Question: what is the reason to remove this statement?

Reply:
If initiated by user (e.g. by issuing `netdataregister`), `ClearResubmitDelayTimer` is already done when addinng/removing prefix or service, no necessary to do it here;
If triggered by hearing MLE advertisement, it should be fine to `ClearResubmitDelayTimer()` when network data is consistent (https://github.com/openthread/openthread/pull/4485/files#diff-e518b44189ad2bc7319b004676d0c29fL377), other than in `UpdateRloc()`. Otherwise, it will break the requirement of `DATA_RESUBMIT_DELAY seconds before resending`